### PR TITLE
Dbp 408 part 2 postgres implementation

### PR DESCRIPTION
**Pull request by simoncolincap**
# Description

## Approval for review

- [ ] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

> Notice: Please remove the WIP label if the PR is ready to review, otherwise nobody will review it.

### CODE REVIEW

File: services/Postgres.go [hunk 1]
```diff
@@ -43,8 +44,120 @@ func (postgres PostgresService) GetCycleTimeSeconds() int {
...
+	default:
+		return s.ResourceScalingProposal{}, fmt.Errorf("unsupported scaled object type: %s", object.GetType())
+	}
+	if !cluster.Ready {
+		return s.ResourceScalingProposal{}, fmt.Errorf("cluster %s (%s) is not ready", cluster.ClusterName, cluster.ClusterId)
+	}
+	return postgres.applyRules(*cluster), nil
+}
+
+func (postgres PostgresService) applyRules(cluster s.Cluster) s.ResourceScalingProposal {
```
Comment (dimapin) [inline services/Postgres.go:61]: The name is a bit misleading. It should be named calculateScalingProposal or somethin like that

File: providers/Ionos.go [hunk 2]
```diff
@@ -144,19 +152,115 @@ func (i Ionos) updateServer(server s.Server, scalingProposal s.ResourceScalingPr
 	}
 
 	slog.Info(fmt.Sprintf("Target for server %s: %d cores, %d bytes\n", server.ServerName, *targetServer.Properties.Cores, *targetServer.Properties.Ram))
-	_, _, err = i.Api.ServersApi.DatacentersServersPut(context.TODO(), server.DatacenterId, server.ServerId).Server(targetServer).XContractNumber(int32(i.Config.ContractId)).Execute()
-	if err != nil {
-		errorsTotalCounter.Inc()
-		return fmt.Errorf("error while setting server resources: %s", err)
-	}
+	// _, _, err = i.Api.ServersApi.DatacentersServersPut(context.TODO(), server.DatacenterId, server.ServerId).Server(targetServer).XContractNumber(int32(i.Config.ContractId)).Execute()
```
[the remark this answers is not in the document]
Comment (simoncolincap) [reply providers/Ionos.go]: This was commented out to verify the scaling without making any changes, it's commented back in

File: services/postgres_test.go [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+	cluster.ResourceState.Cpu = &resourceState
+
+	proposal := postgresService.applyRules(cluster)
+	if proposal.Cpu.Direction != expected {
+		t.Fatalf("Expected CPU scale direction to be %s but got %s", expected, proposal.Cpu.Direction)
+	}
+}
+
+// Check that a cluster with below minimum resources is scaled up
+func TestPostgresApplyRulesRule1(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (simoncolincap) [reply services/postgres_test.go:99]: Done, I also added the unit tests to the Confluence documentation

File: services/Postgres.go (hunk 1 above)
Comment (dimapin) [reply services/Postgres.go:48]: computeScalingProposalInternal or computeTargetResources

Comment (simoncolincap) [reply services/Postgres.go]: Done

**Review by dimapin — APPROVED** [review]

**Pull request merged**